ghcide/src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -141,7 +141,10 @@ runLanguageServer options inH outH getHieDbLoc defaultConfig onConfigurationChan
             logInfo (ideLogger ide) $ T.pack $ "Registering ide configuration: " <> show initConfig
             registerIdeConfiguration (shakeExtras ide) initConfig
 
-            _ <- flip forkFinally (const exitClientMsg) $ runWithDb dbLoc $ \hiedb hieChan -> do
+            let handleHieDbException e = do
+                    logError (ideLogger ide) (T.pack $ "Unexpected exception in hiedb thread: " <> show e)
+                    exitClientMsg
+            _ <- flip forkFinally handleHieDbException $ runWithDb dbLoc $ \hiedb hieChan -> do
```

Comment:
This isn't the hiedb thread, but the server thread. The hiedb thread is forked again in `runWithDb` and shouldn't bring down the main server thread.

Reply:
Thanks, I didn't look closely enough. I have fixed the log message to reflect that this is the server thread